File: api_csharp/WebApplication1/Entity/Reservation.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;

namespace Reservation.Entity
{
    public class Reservation : BaseEntity
    {

        public int ReservationId { set; get; }
        public int CustomerId { set; get; }

        public string ReservationDate { set; get; }

        public int TimeFrameId { set; get; }

        // ---

        public Customer Customer { set; get; }
        public TimeFrame TimeFrame { set; get; } 

    }
}

File: ui_react/src/types/Reservation.ts
export type Reservation = {
    customerId: number,
    name: string,
    reservations: [
        {
            reservationId: number,
            customerId: number,
            reservationDate: string,
            timeFrameId: number,
            timeFrame: {
                dayOfWeek: number,
                startTime: string,
                endTime: string,
                frame: {
                    facilityId: number,
                    facility: {   
                        facilityName: string,
                    }
                }
            }
        }
    ],
};

export type ReservationResponse = {
    reservationId: number,
    customerId: number,
    reservationDate: string,
    timeFrameId: number,
    customer: {
        customerId: number,
        name: string,
    },
    timeFrame: {
        dayOfWeek: number,
        startTime: string,
        endTime: string,
        frame: {
            facilityId: number,
            facility: {   
                facilityName: string,
            }
        }
    }
}

export type ReservationForList = {
    customerId: number,
    name: string,
    reservationId: number,
    dayOfWeek: number,    
    startTime: string,
    endTime: string,
    facilityId: number,
    facilityName: string,
}